src/data/data.jl

```diff
@@ -128,11 +128,8 @@ end
 Adapt.adapt_structure(to, data::IJFH{S, Nij}) where {S, Nij} =
     IJFH{S, Nij}(Adapt.adapt(to, parent(data)))
 
-
-function Base.size(data::IJFH)
-    array = parent(data)
-    size(array, 1), size(array, 4)
-end
+Base.length(data::IJFH) = size(parent(data), 4)
+Base.size(data::Data2D) = (length(data),)
```

Comment:
@jakebolewski I changed this to be just the number of elements, since we can now determine the `Nij` from the type space. I think this simplifies things a bit, but let me know what you think.

Reply:
:+1:  to this solution, I think this is clearer.